File: docker_file_without_multistage/calc.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func main() {
    if len(os.Args) < 4 {
        fmt.Println("Usage: calc <num1> <operator> <num2>")
        fmt.Println("Operators: +, -, *, /")
        return
    }

    num1, err1 := strconv.ParseFloat(os.Args[1], 64)
    operator := os.Args[2]
    num2, err2 := strconv.ParseFloat(os.Args[3], 64)

    if err1 != nil || err2 != nil {
        fmt.Println("Please provide valid numbers.")
        return
    }

    switch operator {
    case "+":
        fmt.Printf("%f + %f = %f\n", num1, num2, num1+num2)
    case "-":
        fmt.Printf("%f - %f = %f\n", num1, num2, num1-num2)
    case "*":
        fmt.Printf("%f * %f = %f\n", num1, num2, num1*num2)
    case "/":
        if num2 == 0 {
            fmt.Println("Error: Division by zero.")
        } else {
            fmt.Printf("%f / %f = %f\n", num1, num2, num1/num2)
        }
    default:
        fmt.Println("Invalid operator. Use one of: +, -, *, /")
    }
}

File: docker_file_without_multistage/calc.py
import sys

def main():
    if len(sys.argv) < 4:
        print("Usage: calc.py <num1> <operator> <num2>")
        print("Operators: +, -, *, /")
        return

    try:
        num1 = float(sys.argv[1])
        operator = sys.argv[2]
        num2 = float(sys.argv[3])
    except ValueError:
        print("Please provide valid numbers.")
        return

    if operator == '+':
        print(f"{num1} + {num2} = {num1 + num2}")
    elif operator == '-':
        print(f"{num1} - {num2} = {num1 - num2}")
    elif operator == '*':
        print(f"{num1} * {num2} = {num1 * num2}")
    elif operator == '/':
        if num2 == 0:
            print("Error: Division by zero.")
        else:
            print(f"{num1} / {num2} = {num1 / num2}")
    else:
        print("Invalid operator. Use one of: +, -, *, /")

if __name__ == "__main__":
    main()
